src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -0,0 +1,341 @@
+            {
+                if (_scriptExecutor == null)
+                {
+                    _scriptExecutor = ServiceLocator.GetInstanceSafe<IScriptExecutor>();
+                }
+
+                return _scriptExecutor;
+            }
+        }
+
+        public override async Task<IReadOnlyList<PackageSpec>> GetPackageSpecsAsync(DependencyGraphCacheContext context)
+        {
+            PackageSpec packageSpec;
+            if (context.PackageSpecCache.TryGetValue(_project.ProjectFullPath, out packageSpec))
+            {
+                return new[] { packageSpec };
+            }
+
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            packageSpec = await GetPackageSpec();
+            context.PackageSpecCache.Add(_project.ProjectFullPath, packageSpec);
+            return new[] { packageSpec };
+        }
+
+        public override async Task<bool> ExecuteInitScriptAsync(
+            PackageIdentity identity,
+            string packageInstallPath,
+            INuGetProjectContext projectContext,
+            bool throwOnFailure)
+        {
+            return
+                await
+                    ScriptExecutorUtil.ExecuteScriptAsync(identity, packageInstallPath, projectContext, ScriptExecutor,
+                        _project.DTEProject, throwOnFailure);
+        }
+
+        #endregion
+
+        #region NuGetProject
+
+        public override async Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            return GetPackageReferences(await GetPackageSpec());
+        }
+
+        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity, DownloadResourceResult downloadResourceResult, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        {
+            try
+            {
+                // We don't adjust package reference metadata from UI
+                await _project.AddOrUpdateLegacyCSProjPackageAsync(
+                    packageIdentity.Id,
+                    packageIdentity.Version.ToString(),
+                    new string[] { },
+                    new string[] { });
+            }
+            catch (Exception e)
+            {
+                nuGetProjectContext.Log(MessageLevel.Warning, e.Message, packageIdentity, _project.Name);
+                return false;
+            }
+
+            return true;
+        }
+
+        public override async Task<Boolean> UninstallPackageAsync(PackageIdentity packageIdentity, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        {
+            try
+            {
+                await _project.RemoveLegacyCSProjPackageAsync(packageIdentity.Id);
+            }
+            catch (Exception e)
+            {
+                nuGetProjectContext.Log(MessageLevel.Warning, e.Message, packageIdentity, _project.Name);
```

Comment:
>nuGetProjectContext [](start = 16, length = 19)

no try catch here, it should be at a higher level so all project types share the same try/catch